File: yangxj96-starter/yangxj96-starter-db/src/main/java/io/github/yangxj96/starter/db/autoconfigure/MybatisPlusAutoConfiguration.java
package io.github.yangxj96.starter.db.autoconfigure;

import com.baomidou.mybatisplus.annotation.DbType;
import com.baomidou.mybatisplus.core.handlers.MetaObjectHandler;
import com.baomidou.mybatisplus.extension.plugins.MybatisPlusInterceptor;
import com.baomidou.mybatisplus.extension.plugins.inner.BlockAttackInnerInterceptor;
import com.baomidou.mybatisplus.extension.plugins.inner.PaginationInnerInterceptor;
import com.zaxxer.hikari.HikariDataSource;
import io.github.yangxj96.starter.db.configure.jdbc.MetaObjectHandlerImpl;
import io.github.yangxj96.starter.db.properties.DBProperties;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;

import javax.sql.DataSource;

@Slf4j
@ConditionalOnProperty(name = "yangxj96.db.jdbc-enable", havingValue = "true")
@EnableConfigurationProperties(DBProperties.class)
public class MybatisPlusAutoConfiguration {

    private final DBProperties properties;

    private static final String LOG_PREFIX = "[自动配置-MyBatis] ";

    public MybatisPlusAutoConfiguration(DBProperties properties) {
        this.properties = properties;
    }

    @Bean
    @ConfigurationProperties("spring.datasource")
    public DataSource dataSource() {
        log.debug("{}载入默认数据源", LOG_PREFIX);
        return new HikariDataSource();
    }

    @Bean
    public MetaObjectHandler metaObjectHandler() {
        log.debug("{}元对象字段填充控制器", LOG_PREFIX);
        return new MetaObjectHandlerImpl();
    }

    /**
     * MyBatis Plus插件
     * <p>一缓和二缓遵循mybatis的规则</p>
     *
     * @return {@link MybatisPlusInterceptor} MyBatis Plus拦截器
     */
    @Bean
    public MybatisPlusInterceptor mybatisPlusInterceptor() {
        log.debug(LOG_PREFIX + "载入MybatisPlusInterceptor");
        // 分页插件
        PaginationInnerInterceptor pageInterceptor = new PaginationInnerInterceptor();
        pageInterceptor.setOverflow(true);
        pageInterceptor.setMaxLimit(500L);
        pageInterceptor.setDbType(DbType.POSTGRE_SQL);
        MybatisPlusInterceptor interceptor = new MybatisPlusInterceptor();
        interceptor.addInnerInterceptor(pageInterceptor);
        // 针对 update 和 delete 语句 作用: 阻止恶意的全表更新删除
        interceptor.addInnerInterceptor(new BlockAttackInnerInterceptor());
        return interceptor;
    }

}

File: yangxj96-starter/yangxj96-starter-db/src/main/java/io/github/yangxj96/starter/db/autoconfigure/MybatisPlusAutoConfiguration.kt
package io.github.yangxj96.starter.db.autoconfigure

import com.baomidou.mybatisplus.annotation.DbType
import com.baomidou.mybatisplus.core.handlers.MetaObjectHandler
import com.baomidou.mybatisplus.extension.plugins.MybatisPlusInterceptor
import com.baomidou.mybatisplus.extension.plugins.inner.BlockAttackInnerInterceptor
import com.baomidou.mybatisplus.extension.plugins.inner.PaginationInnerInterceptor
import io.github.yangxj96.starter.db.configure.jdbc.MetaObjectHandlerImpl
import io.github.yangxj96.starter.db.props.DBProperties
import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean

/**
 * mybatis的自动配置类
 */
@ConditionalOnProperty(name = ["yangxj96.db.jdbc-enable"], havingValue = "true", matchIfMissing = true)
@EnableConfigurationProperties(DBProperties::class)
class MybatisPlusAutoConfiguration {

    companion object {
        private const val LOG_PREFIX = "[自动配置-mybatis plus]:"

        private val log = LoggerFactory.getLogger(this::class.java)
    }

    @Bean
    fun metaObjectHandler(): MetaObjectHandler {
        log.debug("{}元对象字段填充控制器", LOG_PREFIX)
        return MetaObjectHandlerImpl()
    }

    /**
     * MyBatis Plus插件
     *
     * 一缓和二缓遵循mybatis的规则
     *
     * @return [MybatisPlusInterceptor] MyBatis Plus拦截器
     */
    @Bean
    fun mybatisPlusInterceptor(): MybatisPlusInterceptor {
        log.debug(LOG_PREFIX + "载入MybatisPlusInterceptor")
        // 分页插件
        val pageInterceptor = PaginationInnerInterceptor()
        pageInterceptor.isOverflow = true
        pageInterceptor.maxLimit = 500L
        pageInterceptor.dbType = DbType.POSTGRE_SQL
        val interceptor = MybatisPlusInterceptor()
        interceptor.addInnerInterceptor(pageInterceptor)
        // 针对 update 和 delete 语句 作用: 阻止恶意的全表更新删除
        interceptor.addInnerInterceptor(BlockAttackInnerInterceptor())
        return interceptor
    }


}
